packaging/os/yum_repository.py

```diff
@@ -0,0 +1,365 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2015, Rushan Shaymardanov <rush.ru@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+DOCUMENTATION='''
+---
+module: yum_repository
+short_description: Add and remove yum repositories
+description:
+    - Add or remove an yum repositories.
+notes:
+    - This module works on any distribution with yum installed.
+options:
+    id:
+        required: true
+        default: none
+        description:
+            - Must be a unique name for each repository, one word.
+    state:
+        required: false
+        choices: [ "absent", "present" ]
+        default: "present"
+        description:
+            - A repository state.
+    name:
+        required: false
```

Comment:
If parameter is not specified, it will not be added to repository configuration and yum will use it's own default for it. If I add defaults for all non-required parameters, module will put them all in config , that is not very good I think.